From be6f4e52a984bfc10ca3cc79bf812223447802ab Mon Sep 17 00:00:00 2001
From: "Alan T. DeKok" <aland@freeradius.org>
Date: Wed, 22 Feb 2017 10:48:51 -0500
Subject: [PATCH] Improve ip/v4/v6/addr documentation

Contains the following commits.

better documentation for ipaddr & friends.  Fixes #1921

(cherry picked from commit 99e08b85b33e27eb0e0f4e870c50caf8fff6d84f)

typo

(cherry picked from commit 81fe1079edcb94f5b810d894ea255cef5d84985d)
---
 raddb/sites-available/default | 39 +++++++++++++++++++++++++++++----------
 1 file changed, 29 insertions(+), 10 deletions(-)

diff --git a/raddb/sites-available/default b/raddb/sites-available/default
index 083407596..e47f19192 100644
--- a/raddb/sites-available/default
+++ b/raddb/sites-available/default
@@ -85,16 +85,35 @@ listen {
 	#  proxy listeners are automatically created.
 
 	#  ipaddr/ipv4addr/ipv6addr - IP address on which to listen.
-	#  Out of several options the first one will be used.
-	#
-	#  Allowed values are:
-	#	IPv4 address (e.g. 1.2.3.4, for ipv4addr/ipaddr)
-	#	IPv6 address (e.g. 2001:db8::1, for ipv6addr/ipaddr)
-	#	hostname     (radius.example.com,
-	#			A record for ipv4addr,
-	#       		AAAA record for ipv6addr,
-	#			A or AAAA record for ipaddr)
-	#       wildcard     (*)
+	#  If multiple ones are listed, only the first one will
+	#  be used, and the others will be ignored.
+	#
+	#  The configuration options accept the following syntax:
+	#
+	#  ipv4addr - IPv4 address (e.g.192.0.2.3)
+	#  	    - wildcard (i.e. *)
+	#  	    - hostname (radius.example.com)
+	#  	      Only the A record for the host name is used.
+	#	      If there is no A record, an error is returned,
+	#	      and the server fails to start.
+	#
+	#  ipv6addr - IPv6 address (e.g. 2001:db8::1)
+	#  	    - wildcard (i.e. *)
+	#  	    - hostname (radius.example.com)
+	#  	      Only the AAAA record for the host name is used.
+	#	      If there is no AAAA record, an error is returned,
+	#	      and the server fails to start.
+	#
+	#  ipaddr   - IPv4 address as above
+	#  	    - IPv6 address as above
+	#  	    - wildcard (i.e. *), which means IPv4 wildcard.
+	#	    - hostname
+	#	      If there is only one A or AAAA record returned
+	#	      for the host name, it is used.
+	#	      If multiple A or AAAA records are returned
+	#	      for the host name, only the first one is used.
+	#	      If both A and AAAA records are returned
+	#	      for the host name, only the A record is used.
 	#
 	# ipv4addr = *
 	# ipv6addr = *
-- 
2.11.0

